✨ Updated Makefile to support go install kustomize

lauchokyip opened the pull request:
This change will change the method of installation from downloading the script to using `go install`
Fixes https://github.com/kubernetes-sigs/kubebuilder/issues/2822

lauchokyip commented:
@camilamacedo86 I couldn't find how this is generated (https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/Makefile#L138-L146) using go `template` , any tips on how to do that? Thanks!

AlmogBaku commented:
/lgtm

lauchokyip commented:
/cc @camilamacedo86 

Can you PTAL? thanks!

varshaprasad96 reviewed (commented):
/lgtm
/ok-to-test

This looks good. @lauchokyip. Thanks for working on this. This would also need an update in the migration guide in the KB docs. In this section to be specific (https://kubebuilder.io/migration/migration_guide_gov3_to_gov4.html?highlight=v3,to,v4#initialize-a-gov4-project). Would you mind taking a look and changing them.

lauchokyip commented:
@varshaprasad96 Thanks! added over [here](https://github.com/kubernetes-sigs/kubebuilder/pull/3398/files#diff-e58929a1be8f8448e35942c1df3c4987a41a39dd7219cf130f2300135d48aa37R126)

varshaprasad96 reviewed (approved):
Looks good to me! Thanks @lauchokyip

@camilamacedo86 @everettraven @Kavinjsir can we have another lgtm to merge this?

lauchokyip commented:
/retest

camilamacedo86 commented on docs/book/src/component-config-tutorial/testdata/project/Makefile:
```diff
@@ -141,15 +141,14 @@ ENVTEST ?= $(LOCALBIN)/setup-envtest
 KUSTOMIZE_VERSION ?= v5.0.1
 CONTROLLER_TOOLS_VERSION ?= v0.12.0
 
-KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
```
Can you please have your branch rebased with master and run `make generate` to update the samples and docs?
Also, please when you push the PR to update this one can you have the commits squashed?

lauchokyip replied:
My apology, I forgot to setup `upstream` using `git remote`

camilamacedo86 replied:
all fine. Thank you

lauchokyip commented:
@camilamacedo86 , can you run `make generate` on this branch? I kept getting this 
```
$ git status
On branch kuztomize-go-install
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   docs/book/src/component-config-tutorial/testdata/project/config/crd/kustomization.yaml
        modified:   docs/book/src/cronjob-tutorial/testdata/project/config/crd/kustomization.yaml
```

camilamacedo86 reviewed (approved):
/lgtm
